pkg/serverless/logs/logs_collector.go

```diff
@@ -187,8 +195,8 @@ func (lc *LambdaLogsCollector) processMessage(
 
 	if lc.enhancedMetricsEnabled {
 		tags := tags.AddColdStartTag(lc.extraTags.Tags, lc.lastRequestID == lc.coldstartRequestID)
-		if message.logType == logTypeFunction {
-			serverlessMetrics.GenerateEnhancedMetricsFromFunctionLog(message.stringRecord, message.time, tags, lc.demux)
+		if message.logType == logTypeFunction && !lc.outOfMemory {
+			lc.outOfMemory = serverlessMetrics.GenerateEnhancedMetricsFromFunctionLog(message.stringRecord, message.time, tags, lc.demux)
```

Comment:
As you've implemented it, `GenerateEnhancedMetricsFromFunctionLog` is doing several things: determining if an OOM occurred, generating metrics, telling the caller if an OOM occurred.

Instead, I think we should separate concerns into to functions. One function will tell us if there was an OOM and the second will generate the metrics.

I think this is important because, from just looking at the `GenerateEnhancedMetricsFromFunctionLog` code, I have a lot of questions. Why it is returning a `bool`? What does that `bool` mean? If I update the code, will the functionality of this `bool` change? I have no idea.

Creating a separate func to handle just the `bool` part allows for two things: first it helps the reader better understand what the `bool` means and is used for and second it makes the code easier to unit test.

Reply:
Solid points 👍 I broke this out into a separate function, but kept the tests about the same